### PR TITLE
Add psalm-mutation-free to ext-ds

### PR DESCRIPTION
**Pull request by enumag**
I noticed that adding `@psalm-mutation-free` to `Collection::isEmpty()` doesn't affect `Set::isEmpty()` even though `Set implements Collection`. Do I have to copy the methods in such cases or is there a way to make the inheritance work?

### CODE REVIEW

**Comment by muglug** [comment]
You have to copy the methods – `@psalm-mutation-free` and similar are not inherited

**Comment by enumag** [comment]
@muglug Okay, all done.

File: stubs/ext-ds.php [hunk 1]
```diff
@@ -462,37 +598,80 @@ public function copy(): Pair
...
+     * @return Sequence<TKey, TValue>
+     * @psalm-mutation-free
+     */
+    public function copy(): Sequence;
+
+    /**
+     * @return array<TKey, TValue>
+     * @psalm-mutation-free
+     */
+    public function toArray(): array;
```
Comment (enumag) [inline stubs/ext-ds.php:611]: @muglug The failing tests are not related in my opinion. However I found a different error - TKey type doesn't exists in Sequence. This tells me that there is no test that the stubs are valid which I think there should be.

```suggestion
    /**
     * @return Sequence<int, TValue>
     * @psalm-mutation-free
     */
    public function copy(): Sequence;

    /**
     * @return array<int, TValue>
     * @psalm-mutation-free
     */
    public function toArray(): array;
```

**Comment by muglug** [comment]
@enumag the stubs in general are tested, but I don't think the `ext-ds` ones are because they're only loaded conditionally.

**Comment by enumag** [comment]
Could you somehow manually test them? Just to check if I didn't mess up something...

**Pull request merged**